File: ports/rust/src/core/io/bcsfile.rs
use std::fs;
use std::path::Path;

// BcsFile abstracts standard file I/O operations for Rust
pub struct BcsFile {
    path: String,
}

impl BcsFile {
    pub fn new(path: &str) -> Self {
        BcsFile {
            path: path.to_string(),
        }
    }

    pub fn exists(&self) -> bool {
        Path::new(&self.path).is_file()
    }

    pub fn read_all(&self) -> std::io::Result<String> {
        fs::read_to_string(&self.path)
    }

    pub fn write_all(&self, payload: &str) -> std::io::Result<()> {
        fs::write(&self.path, payload)
    }
}

File: ports/go/src/core/io/bcsfile.go
package io

import (
	"os"
)

// BcsFile abstracts standard file I/O operations
type BcsFile struct {
	path string
	file *os.File
}

func NewBcsFile(path string) *BcsFile {
	return &BcsFile{path: path}
}

// Exists checks if the file is present on the filesystem
func (f *BcsFile) Exists() bool {
	info, err := os.Stat(f.path)
	if os.IsNotExist(err) {
		return false
	}
	return !info.IsDir()
}

// ReadAll loads the entire file into a string
func (f *BcsFile) ReadAll() (string, error) {
	bytes, err := os.ReadFile(f.path)
	if err != nil {
		return "", err
	}
	return string(bytes), nil
}

// WriteAll writes the payload string to the file
func (f *BcsFile) WriteAll(payload string) error {
	return os.WriteFile(f.path, []byte(payload), 0644)
}
